admob/src/android/ad_request_converter.cc

```diff
@@ -194,5 +137,38 @@ AdRequestConverter::~AdRequestConverter() {
 
 jobject AdRequestConverter::GetJavaRequestObject() { return java_request_; }
 
+void AdRequestConverter::ConvertRequestConfiguration(AdRequest request) const {
+  JNIEnv* env = ::firebase::admob::GetJNI();
+  jobject builder = env->NewObject(request_config_builder::GetClass(),
+                                   request_config_builder::GetMethodId(
+                                       request_config_builder::kConstructor));
+  // Test DeviceIds
+  if (request.test_device_id_count > 0) {
+    std::vector<std::string> test_devices_vector;
+    for (int i = 0; i < request.test_device_id_count; i++) {
+      test_devices_vector.push_back(request.test_device_ids[i]);
+    }
+    jobject test_device_list =
+        util::StdVectorToJavaList(env, test_devices_vector);
+    builder = util::ContinueBuilder(
+        env, builder,
+        env->CallObjectMethod(builder,
+                              request_config_builder::GetMethodId(
+                                  request_config_builder::kSetTestDeviceIds),
+                              test_device_list));
+    env->DeleteLocalRef(test_device_list);
+
+    // Build request configuration.
+    jobject request_configuration = env->CallObjectMethod(
```

Comment:
Does request_configuration need to be deleted?

Reply:
Good catch.  Please Re-review.

Reply:
FWIW this method is removed in the [next PR](https://github.com/firebase/firebase-cpp-sdk/pull/665/files).  Not that we should ever check-in code that leaks memory...